src/cargo/sources/path.rs

```diff
@@ -246,6 +259,52 @@ impl<'gctx> PathSource<'gctx> {
         Ok(None)
     }
 
+    /// Returns [`Some(gix::Repository)`](gix::Repository) if there is a sibling `Cargo.toml` and `.git`
+    /// directory; otherwise, the caller should fall back on full file list.
+    fn discover_gix_repo(&self, root: &Path) -> CargoResult<Option<gix::Repository>> {
+        let repo = match gix::discover(root) {
```

Comment:
Do we need something like https://github.com/rust-lang/cargo/blob/69726309bf12443a618643cc86125ef60dcbecfb/src/bin/cargo/main.rs#L345-L365 ?

Reply:
A great catch! 

And I think it makes sense to at least open it so that it's isolated and ignores user and system configuration, just so that it aligns with the `gix::open_opts()` when opening submodules.

`gix` implements its security mechanism differently as it keeps track of the trust-level of each key. Code that wants to deal with paths/scripts will not see values that come from untrusted files. Thus one can interact with such repositories ordinarily, but without risking to execute untrusted programs.